Sources/Subprocess/IO/AsyncIO+Darwin.swift

```diff
@@ -0,0 +1,165 @@
+            var buffer: DispatchData = .empty
+            dispatchIO.read(
+                offset: 0,
+                length: maxLength,
+                queue: .global()
+            ) { done, data, error in
+                if error != 0 {
+                    continuation.resume(
+                        throwing: SubprocessError(
+                            code: .init(.failedToReadFromSubprocess),
+                            underlyingError: .init(rawValue: error)
+                        )
+                    )
+                    return
+                }
+                if let data = data {
+                    if buffer.isEmpty {
+                        buffer = data
+                    } else {
+                        buffer.append(data)
+                    }
+                }
+                if done {
+                    if !buffer.isEmpty {
+                        continuation.resume(returning: buffer)
+                    } else {
+                        continuation.resume(returning: nil)
+                    }
+                }
+            }
+        }
+    }
+
+    #if SubprocessSpan
+    internal func write(
+        _ span: borrowing RawSpan,
+        to diskIO: borrowing IOChannel
+    ) async throws -> Int {
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Int, any Error>) in
+            let dispatchData = span.withUnsafeBytes {
+                return DispatchData(
+                    bytesNoCopy: $0,
+                    deallocator: .custom(
+                        nil,
+                        {
+                            // noop
+                        }
+                    )
+                )
+            }
+            self.write(dispatchData, to: diskIO) { writtenLength, error in
+                if let error = error {
+                    continuation.resume(throwing: error)
+                } else {
+                    continuation.resume(returning: writtenLength)
+                }
+            }
+        }
+    }
+    #endif  // SubprocessSpan
+
+    internal func write(
+        _ array: [UInt8],
+        to diskIO: borrowing IOChannel
+    ) async throws -> Int {
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Int, any Error>) in
+            let dispatchData = array.withUnsafeBytes {
+                return DispatchData(
+                    bytesNoCopy: $0,
+                    deallocator: .custom(
+                        nil,
+                        {
+                            // noop
+                        }
+                    )
+                )
+            }
+            self.write(dispatchData, to: diskIO) { writtenLength, error in
+                if let error = error {
+                    continuation.resume(throwing: error)
+                } else {
+                    continuation.resume(returning: writtenLength)
+                }
+            }
+        }
```

Comment:
Same problem here, with the same kind of rearrangement as the solution:

```suggestion
            array.withUnsafeBytes {
                let dispatchData = DispatchData(
                    bytesNoCopy: $0,
                    deallocator: .custom(
                        nil,
                        {
                            // noop
                        }
                    )
                )
                self.write(dispatchData, to: diskIO) { writtenLength, error in
                    if let error {
                        continuation.resume(throwing: error)
                    } else {
                        continuation.resume(returning: writtenLength)
                    }
                }
            }
        }
```